### PR TITLE
Pull request for async-http-client-part2

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: mergify_engine/tests/functional/test_engine.py [hunk 1]
```diff
@@ -177,7 +175,10 @@ def test_debugger(self):
 
         self.setup_repo(yaml.dump(rules))
         p, _ = self.create_pr()
-        debug.report(p.html_url)
+
+        # NOTE(sileht): Run is thread due to not mess the main asyncio loop
```
Comment (jd) [inline mergify_engine/tests/functional/test_engine.py:179]:
```suggestion
        # NOTE(sileht): Run in a thread to not mess with the main asyncio loop
```

File: mergify_engine/tests/functional/actions/test_request_reviews.py [hunk 2]
```diff
@@ -82,7 +82,11 @@ def test_request_reviews_already_max(self):
                 {
                     "name": "request_reviews",
                     "conditions": [f"base={self.master_branch_name}"],
-                    "actions": {"request_reviews": {"users": ["mergify-test1"]}},
+                    "actions": {
+                        "request_reviews": {
+                            "users": ["mergify-test1", "mergify-test-2"]
```
[the remark this answers is not in the document]
Comment (sileht) [reply mergify_engine/tests/functional/actions/test_request_reviews.py:87]: I will move out all tests changes.

File: mergify_engine/utils.py [hunk 3]
```diff
@@ -107,6 +108,18 @@ def get_pull_logger(pull):
     )
 
 
+async def async_main(*awaitables):
+    # NOTE(sileht): This looks useless but
+    #   asyncio.run(asyncio.gather(...))
+    # does not work, because when the gather() coroutine is created, it need the current
+    # loop but asyncio.run() haven't create it yet, so it doesn't work.
```
Comment (jd) [inline mergify_engine/utils.py:115]: 👍

Comment (jd) [inline mergify_engine/utils.py:115]:
```suggestion
    # loop but asyncio.run() haven't created it yet, so it doesn't work.
```

File: mergify_engine/utils.py [hunk 4]
```diff
@@ -107,6 +108,18 @@ def get_pull_logger(pull):
     )
 
 
+async def async_main(*awaitables):
+    # NOTE(sileht): This looks useless but
+    #   asyncio.run(asyncio.gather(...))
+    # does not work, because when the gather() coroutine is created, it need the current
```
Comment (jd) [inline mergify_engine/utils.py:114]:
```suggestion
    # does not work, because when the gather() coroutine is created, it needs the current
```

**Pull request merged**